apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -346,18 +331,40 @@ export class SendMessage {
       return false;
     }
 
-    const { preference } = await this.getSubscriberTemplatePreferenceUsecase.execute(
-      GetSubscriberTemplatePreferenceCommand.create({
-        organizationId: job._organizationId,
-        subscriberId: subscriber.subscriberId,
-        environmentId: job._environmentId,
-        template,
-        subscriber,
-        tenant: job.tenant,
-      })
-    );
+    let subscriberPreference: { enabled: boolean; channels: IPreferenceChannels };
+    if (command.statelessPreferences) {
+      /*
+       * Stateless Workflow executions do not have their definitions stored in the database.
```

Comment:
@rifont Would it be more helpful to apply the same strategy when it comes to stored workflows by adapting the event trigger usecase and fetching their preferences from the database, before triggering the `send-message.usecase`? 

In other words, always inject preferences to `send-message` and all the downstream use-cases.

Reply:
Yes, it would be more helpful. There's a question of whether we want to make such a change now, or whether we can live with intentional naming that highlights this shortcoming for the interim.